src/util/validateJson.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint @typescript-eslint/no-explicit-any: "off" */
+/* eslint @typescript-eslint/explicit-module-boundary-types: "off" */
+import type {
+  FeatureCollection,
+  Feature,
+  GeoJsonProperties,
+  Geometry
+} from 'geojson';
+
+export function isFeatureCollection(object: any): object is FeatureCollection {
+  return (
+    typeof object !== 'undefined' &&
+    typeof object === 'object' &&
+    object.type === 'FeatureCollection' &&
+    Array.isArray(object.features) &&
+    object.features.every(isFeature)
+  );
+}
+
+export function isFeature(object: any): object is Feature {
+  return (
+    typeof object !== 'undefined' &&
+    typeof object === 'object' &&
+    object.type === 'Feature' &&
+    (typeof object.id === 'string' ||
+      typeof object.id == 'number' ||
+      typeof object.id === 'undefined') &&
+    isGeoJsonProperties(object.properties) &&
+    isGeometry(object.geometry)
+  );
+}
+
+export function isGeoJsonProperties(object: any): object is GeoJsonProperties {
+  return typeof object === 'object' || object === null;
+}
+
+export function isGeometry(object: any): object is Geometry {
+  return (
+    typeof object !== 'undefined' &&
+    typeof object === 'object' &&
+    (object.type === 'Point' ||
+      object.type === 'MultiPoint' ||
+      object.type === 'LineString' ||
+      object.type === 'MultiLineString' ||
+      object.type === 'Polygon' ||
+      object.type === 'MultiPolygon') &&
+    Array.isArray(object.coordinates) &&
+    (typeof object.coordinates[0] === 'number' ||
+      (Array.isArray(object.coordinates[0]) &&
+        (typeof object.coordinates[0][0] === 'number' ||
+          (Array.isArray(object.coordinates[0][0]) &&
+            typeof object.coordinates[0][0][0] === 'number'))))
+  );
```

Comment:
This will probably be a pain to implement, I know, but the semantics of the `coordinates` array(s) is dependent on the the `type` of the geometry, and we should probably validate this.

e.g. 
* `Point` is a 1-D array,
* `MultiPoint` and `LineString` are 2-D arrays
* `MultiLineString` and `Polygon` are 3-D arrays
* `MultiPolygon` are 4-D arrays (which don't actually appear to be checked here!)

Reply:
Replaced with GeoJSON schema validation.